docs/customization/api.rst

```diff
@@ -96,3 +106,346 @@ The next step is to modify the security configuration in ``config/packages/secur
+
+    You can create your own serialization group for every endpoint or use the one out of the box. If you don't know the name of group for endpoint you want to modify,
+    you can find it by searching for your class configuration file in `%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources``
+    and look for path that you want to modify.
+
+In this case the new ``group`` is called ``admin:product:read``:
+
+.. code-block:: xml
+
+    <!--...-->
+    <attribute name="averageRating">
+            <group>admin:product:read</group>
+            <group>shop:product:read</group>
+    </attribute>
+    <!--...-->
+
+After this change your response should be extended with new field:
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "averageRating": 3,
+        //...
+    }
+
+.. tip::
+
+    Read more about API Platform `serialization groups <https://api-platform.com/docs/core/serialization/#using-serialization-groups>`_
+
+
+We were able to add a field that exists in ``Product`` class, but what if you want to extend it with custom fields?
+Let's customize response now with your custom fields serialized in response.
+
+Adding a custom field to response
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to add a new field named ``additionalText`` to ``Product``.
+First let's create a new serializer that will support our ``Product`` resource:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Serializer;
+
+    use Sylius\Component\Core\Model\ProductInterface;
+    use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+    use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+    use Webmozart\Assert\Assert;
+
+    final class ProductSerializer implements ContextAwareNormalizerInterface
+    {
+        /** @var NormalizerInterface */
+        private $objectNormalizer;
+
+        public function __construct(NormalizerInterface $objectNormalizer) {
+            $this->objectNormalizer = $objectNormalizer;
+        }
+
+        public function normalize($object, $format = null, array $context = [])
+        {
+            Assert::isInstanceOf($object, ProductInterface::class);
+
+            $data = $this->objectNormalizer->normalize($object, $format, $context);
+
+            return $data;
+        }
+
+        public function supportsNormalization($data, $format = null, $context = []): bool
+        {
+            return $data instanceof ProductInterface;
+        }
+    }
+
+And now let's declare its service in config files:
+
+.. code-block:: xml
+
+    <service id="App\Serializer\ProductSerializer">
+            <argument type="service" id="api_platform.serializer.normalizer.item" />
+            <tag name="serializer.normalizer" />
+    </service>
```

Comment:
I would be for adding the service definition in YAML as it is the standard in our docs